coordinator/src/node.rs

```diff
@@ -208,17 +192,45 @@ enum TradeAction {
     // Reduce,
 }
 
+/// Calculates the accept settlement amount based on the pnl.
+///
+/// The smallest possible result is 0.
+fn calculate_accept_settlement_amount(
+    opening_price: Decimal,
+    closing_price: Decimal,
+    quantity: f64,
+    long_leverage: f64,
+    short_leverage: f64,
+    direction: Direction,
+) -> Result<u64> {
+    let pnl = cfd::calculate_pnl(
+        opening_price,
+        closing_price,
+        quantity,
+        long_leverage,
+        short_leverage,
+        direction,
+    )?;
+
+    let margin_trader = calculate_margin(opening_price, quantity, long_leverage);
+
+    let accept_settlement_amount = (margin_trader as i64).saturating_add(pnl);
```

Comment:
I'm unsure if this saturating add is needed at this stage. It's slightly safer, but are we really gonna hit `i64::MAX`? I would just add the PNL to make it more readable.

---

For reference:

```rust
i64::MAX == 9_223_372_036_854_775_807
```